active-directory-wpf-msgraph-v2/MainWindow.xaml.cs

```diff
@@ -37,9 +42,20 @@ private async void CallGraphButton_Click(object sender, RoutedEventArgs e)
             ResultText.Text = string.Empty;
             TokenInfoText.Text = string.Empty;
 
-            var accounts = await app.GetAccountsAsync();
-            var firstAccount = accounts.FirstOrDefault();
+            IAccount firstAccount;
 
+            // WAM will always get an account in the cache. So if we want
+            // to have a chance to select the accounts interactively, we need to
+            // force the non-account
+            if (howToSignIn.SelectedIndex == 1)
+            {
+                firstAccount = null;
+            }
+            else
+            {
+                var accounts = await app.GetAccountsAsync();
```

Comment:
This (`firstOrDefault`) is a bit artificial. In my test apps, I have a drop-down control with all known accounts and users can either select a "known" account or they select none. I think this is closer to real-world scenario than "randomly pick an account". I can contribute this if you wish to go down this path.